activerecord/lib/active_record/connection_adapters/abstract/query_comment.rb

```diff
@@ -0,0 +1,209 @@
+          # Resets the cached comment if <tt>cache_query_comment</tt> is +true+.
+          def update(ctx)
+            return unless ctx.is_a? Hash
+            self.context.merge!(ctx.symbolize_keys)
+            self.cached_comment = nil
+          end
+
+          def comments_available?
+            self.components.present? || self.inline_annotations.present?
+          end
+
+          # Returns a +String+ containing the component-based query comment.
+          # Sets and returns a cached comment if <tt>cache_query_comment</tt> is +true+.
+          def comment
+            return uncached_comment unless cache_query_comment
+            return cached_comment unless cached_comment.nil?
+            self.cached_comment = uncached_comment
+          end
+
+          def uncached_comment
+            "/*#{escape_sql_comment(comment_content)}*/"
+          end
+
+          # Returns a +String+ containing any inline comments from +with_annotation+.
+          def inline_comment
+            return nil unless inline_annotations.present?
+            "/*#{escape_sql_comment(inline_comment_content)}*/"
+          end
+
+          # Manually clear the comment cache.
+          def clear_comment_cache!
+            self.cached_comment = nil
+          end
+
+          # Annotate any query within `&block`. Can be nested.
+          def with_annotation(comment, &block)
+            self.inline_annotations.push(comment)
+            block.call if block.present?
+          ensure
+            self.inline_annotations.pop
+          end
+
+          # Return the set of active inline annotations from +with_annotation+.
+          def inline_annotations
+            context[:inline_annotations] ||= []
+          end
+
+          # QueryComment +component+ methods
+
+          # Set during Rails boot in lib/active_record/railtie.rb
+          def application # :nodoc:
+            context[:application_name]
+          end
+
+          def pid # :nodoc:
+            Process.pid
+          end
+
+          def connection_config # :nodoc:
+            ActiveRecord::Base.connection_db_config
+          end
+
+          def socket # :nodoc:
+            connection_config.socket
+          end
+
+          def db_host # :nodoc:
+            connection_config.host
+          end
+
+          def database # :nodoc:
+            connection_config.database
+          end
+
+          def line # :nodoc:
+            backtrace_cleaner.add_silencer { |line| line.match?(/lib\/active_(record|support)/) }
+            backtrace_cleaner.clean(caller.lazy).first
+          end
+
+          private
+            def context
+              Thread.current[:active_record_query_comment_context] ||= {}
+            end
+
+            def escape_sql_comment(str)
+              return str unless str.include?("*")
+              while str.include?("/*") || str.include?("*/")
+                str = str.gsub("/*", "").gsub("*/", "")
+              end
+              str
+            end
```

Comment:
Do we own the string at this point? Could we do this?

```suggestion
            def escape_sql_comment(comment)
              comment.gsub!("/*", "")
              comment.gsub!("*/", "")
              comment
            end
```

Reply:
We have the same feature as `sanitize_as_sql_comment`. #35660

Reply:
I've copied over the regexp approach from `sanitize_as_sql_comment` - because it's operating on the connection it was getting into a recursive call due to the prepend.

This could be avoided if the method was on the class, instead of the connection instance - could we consider changing that?